### PR TITLE
Feature/initial work

### PR DESCRIPTION
**Pull request by Matt-Guest**
The api router, that proxies any request onto a number of services; 
The supported services are;
* hierarchy api
* filter api
* dataset api
* codelist api
* recipe api 
* import api
* search api

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	GracefulShutdown time.Duration `envconfig:"SHUTDOWN_TIMEOUT"`
+}
+
+var configuration *Config
+
+// Get configures the application and returns the configuration
+func Get() (*Config, error) {
+	if configuration == nil {
+		configuration = &Config{
+			BindAddr:         ":23200",
```
Comment (daiLlew) [inline config/config.go]: There has been some discussion on if we should use defaults or not given that they cause lots of pain in prod. Wider team question: Are we going to abandon them or continue to use them?

File: config/config.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+			FilterAPIURL:     "http://localhost:22100",
+			DatasetAPIURL:    "http://localhost:22000",
+			CodelistAPIURL:   "http://localhost:22400",
+			RecipeAPIURL:     "http://localhost:22300",
+			ImportAPIURL:     "http://localhost:21800",
+			SearchAPIURL:     "http://localhost:23100",
+			GracefulShutdown: 5 * time.Second,
+		}
+		if err := envconfig.Process("", configuration); err != nil {
+			return nil, err
```
Comment (daiLlew) [inline config/config.go]: can you add some context to the error? 'something like config load error' - makes it easier to see what the cause is from the logs.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	addHandler(router, filter, "/filter-outputs")
+	addHandler(router, hierarchy, "/hierarchies")
+	addHandler(router, search, "/search")
+
+	// Private APIs
+	recipe := newAPIProxy(cfg.RecipeAPIURL)
+	importAPI := newAPIProxy(cfg.ImportAPIURL)
+	addHandler(router, recipe, "/recipes")
+	addHandler(router, importAPI, "/jobs")
+
```
Comment (daiLlew) [inline main.go]: You need to add the signal intercept & graceful shutdown stuff - follow the same pattern at the other services.

File: config/config.go (hunk 1 above)
Comment (Matt-Guest) [reply config/config.go]: We should try arranging a team discussion about defaults this week

File: config/config.go (hunk 2 above)
Comment (Matt-Guest) [reply config/config.go]: done

File: main.go (hunk 3 above)
Comment (Matt-Guest) [reply main.go]: The go-ns library has graceful shutdown built in. Its needed for the other services as they need to shutdown kafka or a database. For this services it does not use any other resources.

**Pull request merged**